### PR TITLE
Show condition messages on workspace startup

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR improves the workspace startup page to show DevWorkspace status conditions as part of the creation and loading flow.

#### Screencasts

<details>
<summary>Creating a new workspace</summary>

https://github.com/eclipse-che/che-dashboard/assets/16220722/4fb64791-73a6-4776-8f4a-f05807a7e859

</details>

<details>
<summary>Starting an existing workspace</summary>

https://github.com/eclipse-che/che-dashboard/assets/16220722/8771e27a-48c0-4663-9067-77e681c3db48

</details>

<details>
<summary>Workspace failed to start</summary>

https://github.com/eclipse-che/che-dashboard/assets/16220722/0744f102-3d9f-4bdf-88e5-02c8379d55bf

</details>

### What issues does this PR fix or reference?

resolves https://github.com/eclipse/che/issues/22159

### Is it tested? How?

1. Deploy Eclipse Che
2. Update CheCluster CR to use **quay.io/eclipse/che-dashboard:pr-841** as the dashboard image:
	```yaml
	spec:
	  components:
	    dashboard:
	      deployment:
	        containers:
	          - image: quay.io/eclipse/che-dashboard:pr-841
	```	
4. Start any existing workspace or create a new one.
5. As soon as the workspace gets the status `STARTING` you should be able to see the DevWorkspace condition messages as sub-steps of the `Waiting for workspace to start` step.

### CODE REVIEW

**Comment by akurinnoy** [comment]
@ibuziuk @olexii4 I've cherry-picked https://github.com/eclipse-che/che-dashboard/pull/836 into this PR ([fix: cherry-pick](https://github.com/eclipse-che/che-dashboard/pull/841/commits/ce045eb71a3a4adc63e81f2363c74fbe3d3473eb)) to avoid complicated merge conflicts. Don't you mind closing #836 in favor of this PR?

**Review by ibuziuk — APPROVED** [review]

File: packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Apply/Devfile/index.tsx [hunk 1]
```diff
@@ -229,52 +231,81 @@ class StepApplyDevfile extends AbstractLoaderStep<Props, State> {
...
+
+    // the user devfile is invalid and caused creation error
+    // so we have to proceed with the default devfile
+    if (continueWithDefaultDevfile === true) {
+      if (defaultDevfile === undefined) {
+        throw new Error('Failed to resolve the default devfile.');
+      }
+
+      if (devfile === undefined) {
+        const _devfile = cloneDeep(defaultDevfile);
```
Comment (olexii4) [inline packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Apply/Devfile/index.tsx:265]: The ```cloneDeep``` selector will return a new instance each time:

```
        const devfile = load(devfileContent) as devfileApi.Devfile;
        // propagate default components
        if (!devfile.components || devfile.components.length === 0) {
          devfile.components = defaultComponents;
        }
        return devfile;
```

https://github.com/eclipse-che/che-dashboard/blob/loader-steps-conditions/packages/dashboard-frontend/src/store/DevfileRegistries/selectors.ts#L81

Why do we need this ```cloneDeep(defaultDevfile);```?

**Comment by olexii4** [comment]
I found a bug. 
Знімок екрана 2023-06-22 о 12 46 52

I just tried to check a flow when we created a new workspace with the default devfile:

```https://devspaces.apps.cluster-xr5nk.xr5nk.sandbox2114.opentlc.com/#https://github.com/olexii4/use-reference-editor```

File: packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Apply/Devfile/index.tsx (hunk 1 above)
Comment (akurinnoy) [reply packages/dashboard-frontend/src/components/WorkspaceProgress/CreatingSteps/Apply/Devfile/index.tsx:265]: I think it makes sense to get rid of `cloneDeep`, thanks.

**Comment by akurinnoy** [comment]
@olexii4 I've fixed the bug, can you test the fix?

**Comment by olexii4** [comment]
LGTM
Знімок екрана 2023-06-23 о 12 06 35

**Review by olexii4 — APPROVED** [review]

**Pull request merged**